README.md

```diff
@@ -132,6 +140,38 @@ program. You'll still need to set the environment for `beam_notify` to work. On
 the bright side, this will skip out having your system start `bash` on each
 notification.
 
+## Restricted shell environments
+
+Some programs clear the OS environment before running programs as a security
+precaution. It's still possible send messages to Elixir.
+
+You'll need to know the path to the `beam_notify` binary and have a place to put
+the communications socket that both Elixir and the `beam_notify` binary can
+open. In this example, the socket will be created as
+`/tmp/my_beam_notify_socket`. In Elixir, the `BEAMNotify` child_spec might look
+like this:
+
+```elixir
+{BEAMNotify, path: "/tmp/my_beam_notify_socket", dispatcher: &Some.function/2}
```

Comment:
Do you think we should add the `:name` option for completeness?

Reply:
Good point. The `:name` is still used for the `GenServer` name.